fix(helm): Display debug release info when upgrade failed because of timeout. Closes #3020

splisson opened the pull request:
Als fixes test in runReleaseCases

Commit:
build helm binary in tiller image

adshmh committed:
added description field to api types UpdateReleaseRequest, InstallReleaseRequest, UninstallReleaseRequest, and RollbackReleaseRequest

Signed-off-by: Arash Deshmeh <adeshmeh@ca.ibm.com>

adshmh committed:
added ability to handle custom release descriptions to tiller

Signed-off-by: Arash Deshmeh <adeshmeh@ca.ibm.com>

adshmh committed:
feat(helm): Added the ability to handle custom description to client side for install, upgrade, rollback, and delete commands. Updated the documentation according to the changes.

Signed-off-by: Arash Deshmeh <adeshmeh@ca.ibm.com>

databus23 committed:
Fix --tiller-namespace flag for plugins

This fixes using `--tiller-namespace $namespace` flag (without the equal sign) for helm plugins.

adamreese committed:
fix(kube): output internal object table

fixes #3937

maratoid committed:
add --col-width to `helm search` (#3949)

* add customizeable --col-width

* make docs. whitespacing

BarryWilliams committed:
Changed whitespacing in comments

jstoja committed:
Add quoting support in ingress to allow wildcard domain

eyalbe4 committed:
Fix for - Downloader plugins not used when downloading new repo's index.yaml #3938

jstoja committed:
Change tiller's Dockerfile to use USER nobody + upgrades to alpine:3.7

Commit:
remove need for type reflection

jstoja committed:
docs(helm): update Globs examples to work correctly

Commit:
add amendments to release checklist

Commit:
#3763 replace backslash with forward slash

Commit:
Revert "toYaml - Fix #3470 and #3410's trailing \n issues"

Commit:
Revert "Fix tiller deployment on RBAC clusters"

Commit:
Refactor to be more precise

AdamDang committed:
Typo fix: evalutes->evaluates

AdamDang committed:
Typo fix: usa helm->use helm

usa helm->use helm

thomastaylor312 committed:
docs(release_checklist): Adds information about new release meetings

We have decided to start having release meetings to define clear objectives
for a release. This documents the process.

technosophos committed:
feat(tiller): support CRD installation (#3982)

This adds support for installing CRDs well before any other resource
kinds are installed.

This PR introduces a new hook, `crd-install`, that fires before
manifests are even validated. It is used to install a CRD before any
other part of a chart is installed.

Currently, this hook is _only implemented for install_. That means we
currently cannot add new CRDs during `helm upgrade`, nor can they
be rolled back. This is the safest configuration, as the update/rollback
cycle gets very challenging when CRDs are added and removed.

gabrielsvinha committed:
Upgrade moniker version (#4034)

Commit:
chore(docs): mv index.md to README.md

* Newcomers may not realize they have to look
for the index.md file when coming to the docs dir
* README.md makes the index easier to see

adamreese committed:
fix(kube): run schema validation on BuildUnstructured

winjer committed:
Support values files on remote servers with self-signed certs / client cert authentication (#4003)

* support tls for values file access

- uses --ca-file / --cert-file / --key-file

* also for upgrade command

rajatjindal committed:
fix lint warning

flimzy committed:
Suppress 'unauthenticated users' warning when --tiller-tls-verify present

And clarify the warning to indicate exactly how to correct the problem.

Commit:
replace with a link to the latest releases page

This removes a step required every time we release Helm, making it
simpler to cut a new release.

AdamDang committed:
Typo fix: mach->match

mach->match

jmeickle committed:
Don't request Tiller version in installer

This makes the script hang for quite a while if it can't connect to Tiller to retrieve this information that isn't being used anyways.

AdamDang committed:
Update capabilities.go

rocky-nupt committed:
Fix(helm): fix the bug of the charts not deployed after downloaded
in helm install --dep-up

helm install --dep-up does not deploy the charts downloaded by itself.
I reload the charts path after downloading the missing charts.

Closes #3423

adshmh committed:
feat(helm): added new helm notes command to display notes provided by the chart of a release

Signed-off-by: Arash Deshmeh <adeshmeh@ca.ibm.com>

Commit:
Migrate 'template' to 'include' in 'helm create' to conform to Helm best practices

Signed-off-by: Alejandro Vicente Grabovetsky <sasha@aid.technology>

yenicapotediaz committed:
removed --install option from --namespace.

qinix committed:
fix charts doc: extra space in requirement condition

balboah committed:
Add helm template --is-upgrade

Useful for toggling .Release.IsInstall / .Release.IsUpgrade which
in turn may cause templates to render differently.

balboah committed:
make docs

recipe for target 'verify-docs' failed

goshlanguage committed:
adding docs for GKE for helm 2

slayerjain committed:
docs(helm): Added double quotes to example

In the current version of helm (2.8.2), there needs to be double quotes around the value to support escaping special characters.

mf-lit committed:
Update tiller_ssl.md

Add -days parameter to openssl signing commands for clarity. Also add some related troubleshooting steps.

Commit:
add SECURITY_CONTACTS

Commit:
add every release artifact to the release notes

Commit:
update security vulnerability email

mattjmcnaughton committed:
Fix inaccurate comment in `tiller`

When reading through `cmd/tiller/tiller.go`, I noticed a comment around
`rootServer` mentions the usage of an `init` function. However, there is
no `init` function in this package. Update the comment to be more
accurate.

mattjmcnaughton committed:
Fix concurrency issues with helm install

Address a race condition that arises when running two `helm install`
commands, both of which specify a namespace that does not already exist.

In this specific instance, attempting to create a `namespace` which
already exists shouldn't be a failure which causes `helm install` to
terminate.

radbaron committed:
Documentation: add syntax highlighting to code examples & fix spelling of kube primitives

Commit:
Parse booleans and null as string values with --set-string flag

Commit:
Add test to make sure --set-string flag takes `null` as string

Commit:
Add test to make sure --set flag interprets `null` as `nil`

rsalmond committed:
small typo fix

mattjmcnaughton committed:
Fix test failure message

The error message we reported if the test failed was incorrect.

AdamDang committed:
Typo fix: retruns->returns

retruns->returns

eicnix committed:
Added documentation for the tpl function

mattjmcnaughton committed:
Fix incorrect timestamp when helm package

The files contained in the `tar` file from `helm package` did not have
the correct timestamp. This lack of timestamp occurred because we were
not writing any time to the tar header file.

tariq1890 committed:
Setting DisableCompression to true to disable unwanted httpclient tranformations

mattjmcnaughton committed:
Add containerized options for tests in Makefile

Add an option to run the `test-unit`, `test-style`, and `test`
steps from the `Makefile` insides of a docker container. Doing so
isolates this component of helm development from any other aspect of
your global go environment.

These commands all have the name `docker-*`. Long term, there may
be reproducibility benefits to running all of the Make steps in a
container by default, in which case `containerized-test-unit` could
become `test-unit`.

andrzejwp committed:
Add link to cdwv/awesome-helm

mattjmcnaughton committed:
Increase error message specificity for name reuse

If `helm upgrade` fails because a reused name is still in use, the error
message does not specify which name is still in use. Update the error
message.

mirake committed:
Typo fix: indentifies -> identifies

Commit:
test: add test for 'ParseIntoString' and update test for 'ParseInto'

Commit:
Add s390 architecture in TARGETS

Signed-off-by: Alice Frosi <alice@linux.vnet.ibm.com>

Dean-Coakley committed:
Remove trailing whitespace

liyongxin committed:
typo fix for lengh to length

Commit:
change copyright to "Copyright The Helm Authors"

splisson committed:
fix(helm): Display debug release info when upgrade failed because of timeout. Closes #3020

liyongxin committed:
typo fix for template test

cmattoon committed:
Do not fail linting because of missing 'required' template values

bacongobbler commented on cmd/helm/install.go:
```diff
@@ -278,27 +278,37 @@ func (i *installCmd) run() error {
-	if err != nil {
-		return prettyError(err)
+
+	// If there is an error while waiting, make a call without waiting to get the release content
```
should we be wrapping this in a `if settings.Debug` much like what was done in `helm rollback` and `helm upgrade`?

splisson commented:
I had done that and someone chimed in and told me to remove it
Any way works for me :-)

On Thu, Jun 14, 2018 at 9:10 AM, Matthew Fisher <notifications@github.com>
wrote:

> *@bacongobbler* commented on this pull request.
> ------------------------------
>
> In cmd/helm/install.go
> <https://github.com/kubernetes/helm/pull/4207#discussion_r195483224>:
>
> > @@ -278,27 +278,37 @@ func (i *installCmd) run() error {
>  		helm.InstallDisableCRDHook(i.disableCRDHook),
>  		helm.InstallTimeout(i.timeout),
>  		helm.InstallWait(i.wait))
> -	if err != nil {
> -		return prettyError(err)
> +
> +	// If there is an error while waiting, make a call without waiting to get the release content
>
> should we be wrapping this in a if settings.Debug much like what was done
> in helm rollback and helm upgrade?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/kubernetes/helm/pull/4207#pullrequestreview-128862176>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAILAcSpUMFmCmntASV4sEnyB1Z3zG48ks5t8orjgaJpZM4UjsH2>
> .
>



-- 
Sebastien

munnerz committed:
Add 'name' label to created namespaces

ghugo committed:
chore(sprig): bump sprig to 2.15.0

This change updates sprig to 2.15, which has support for generating
SSL certificates in charts from user defined CA certificates.

jonk committed:
add nil check in if statement configmap example, also note why

0gajun committed:
Change permission of index.yaml

Because index.yaml generated by a command `helm repo index` isn't executable
file, but just configuration file.

Commit:
fix `helm template -x` pathing issues on Windows

tiller's rendering engine converts os filepath separators into unix-style filepath separators, so we need to split template names with a forward slash.

imroc committed:
Support Stdin when using downloaders of plugin

Inject os.Stdin into the command of helm plugin's downloaders

flynnduism committed:
docs(chore): add svg logo

AdamDang committed:
Some small typos:

Some small typos:
Line 92: stategies->strategies
Line 263: namepace->namespace
Line 266: ReuseName->Reuse_name

imroc committed:
docs(helm): add helm-cos to related helm plugins

Commit:
chore(docs): lowercase `Kind` for clarity

`Kind` being upper case makes it seem like we're referrring to
the Kind field of a Kubernetes resource but we're really talking
about the kind of helm hook in this case.
resolves #4302

Commit:
fix(helm): return when listing with no releases or nil

imroc committed:
feat(helm): hiding password input on terminal

When using "helm repo add" with "--username" and without "--password",
hide user's input with a password prompt. This allows users to not
expose their passwords to the command line history.

flynnduism committed:
docs(chore): scale svg up to 620 initial width/height

bjornmagnusson committed:
CONTRIBUTING.md: Corrected slack channel/s in support channels

Commit:
docs(*): update tiller_ssl docs to reflect IP SAN usage.

When using helm/tiller in tls-verify mode, 127.0.0.1 should
be listed as an IP SAN in the tiller certificate to pass
hostname verficiation of the TLS handshake.

Closes #4149

dustyrip committed:
Allow zsh completion to be autoloaded by compinit

adshmh committed:
feat(helm): add ability to sort release list by chart name

Signed-off-by: Arash Deshmeh <adeshmeh@ca.ibm.com>

Commit:
fix(helm): add --tls-hostname flag to tls-flags

fibonacci1729 committed:
Revert "fix(helm): add `--tls-hostname` flag to tls flags."

Commit:
Include exact details to configure storage=secret

Commit:
fix(helm): fix(helm): add `--tls-hostname` flag to tls flags

docs(*): update tiller_ssl.md to reflect IP SAN usage.
When using helm/tiller in tls-verify mode, 127.0.0.1 should
be listed as an IP SAN in the tiller certificate to pass
hostname verficiation of the TLS handshake.

Closes #4149

novas0x2a committed:
[fake] make InstallReleaseFromChart obey DryRun

novas0x2a committed:
[fake] return the correct error message for missing releases

novas0x2a committed:
[templates] extract some rendering code into a package

novas0x2a committed:
[fake] implement rendering and simulated upgrades

consideRatio committed:
helm template cmd - conformity of output (#3811)

By changing a string from `##---` to `---` the output written to files matches the output generated to stdout.

idvoretskyi committed:
Slack channels now have URL's

novas0x2a committed:
[tiller] move the Manifest type to its own pkg

It's really easy to cause an import cycle on this type; this resolves
the problem by moving it out of the tiller pkg into its own. An alias is
left behind in order to prevent downstream breakage.

Commit:
fix(release_server): fix how we merge values

resolves #4337
Merging maps inside of strings gets a bit tricky. When two
strings consisting of "{}" were being added together, this resulted in
"{}\n{}" instead of "{}" which is what we wanted. This led to YAML
parsing errors and showed up when the `--reuse-values` flag was used
when no overrides via `--set` were provided during install and/or
upgrade.

Commit:
ref(docs): add more links to tiller rbac info

Commit:
docs(rbac.md): delete redundant step

dracan committed:
Add missing space

Yeah, I know - very trivial, but ¯\_(ツ)_/¯ :)

rohancme committed:
feat(helm): Add the --kubeconfig flag (#4235)

Adds the ability to specify a kubeconfig file that overrides $KUBECONFIG

mumoshu committed:
feat(helm): Detailed exit code for helm plugins (#4367)

Resolves #4170

idvoretskyi committed:
Snap installation information added (#4366)

* Snap installation information added

Signed-off-by: Ihor Dvoretskyi <ihor@linux.com>

* Moved Snap instructions to the different section

Signed-off-by: Ihor Dvoretskyi <ihor@linux.com>

technosophos committed:
docs(OWNERS): add rimusz as emeritus maintainer (#4357)

This PR was approved by vote of the core maintainers. It adds Rimus, one
of the project's founders, as an emeritus Helm maintainer.

nashasha1 committed:
Dashes are not allowed in the name

mumoshu committed:
feat: Set values from local files (#3758)

Adds the `--set-file key=filepath` flag to `install`, `upgrade`, `template` and `lint` sub-commands so that the content of the file at the `filepath` is set to the value for the `key`.

Resolves #1754

bjornmagnusson committed:
README: updated links to helm/helm

AndersRasmussen committed:
Adding space for correct formatting on docs.helm.sh

The rendering of the bullet list is not rendered correctly on https://docs.helm.sh/developing_charts/#hooks

Commit:
fix path output for Windows

Commit:
soften the recommendation of hypens in chart names

We definitely still suggest that dashes are acceptable for chart packages, but users should be aware of the limitations of dashes in variable names in Go templates.

novas0x2a committed:
[tiller] make update --force --dry-run obey dry-run

rimusz committed:
replace  github org with The Kubernetes package manager

To begin working with Helm, run the 'helm init' command:

	$ helm init

This will install Tiller to your running Kubernetes cluster.
It will also set up any necessary local configuration.

Common actions from this point include:

- helm search:    search for charts
- helm fetch:     download a chart to your local directory to view
- helm install:   upload the chart to Kubernetes
- helm list:      list releases of charts

Environment:
  $HELM_HOME          set an alternative location for Helm files. By default, these are stored in ~/.helm
  $HELM_HOST          set an alternative Tiller host. The format is host:port
  $HELM_NO_PLUGINS    disable plugins. Set HELM_NO_PLUGINS=1 to disable plugins.
  $TILLER_NAMESPACE   set an alternative Tiller namespace (default "kube-system")
  $KUBECONFIG         set an alternative Kubernetes configuration file (default "~/.kube/config")

Usage:
  helm [command]

Available Commands:
  completion  Generate autocompletions script for the specified shell (bash or zsh)
  create      create a new chart with the given name
  delete      given a release name, delete the release from Kubernetes
  dependency  manage a chart's dependencies
  fetch       download a chart from a repository and (optionally) unpack it in local directory
  get         download a named release
  history     fetch release history
  home        displays the location of HELM_HOME
  init        initialize Helm on both client and server
  inspect     inspect a chart
  install     install a chart archive
  lint        examines a chart for possible issues
  list        list releases
  package     package a chart directory into a chart archive
  plugin      add, list, or remove Helm plugins
  repo        add, list, remove, update, and index chart repositories
  reset       uninstalls Tiller from a cluster
  rollback    roll back a release to a previous revision
  search      search for a keyword in charts
  serve       start a local http web server
  status      displays the status of the named release
  template    locally render templates
  test        test a release
  upgrade     upgrade a release
  verify      verify that a chart at the given path has been signed and is valid
  version     print the client/server version information

Flags:
      --debug                           enable verbose output
  -h, --help                            help for helm
      --home string                     location of your Helm config. Overrides $HELM_HOME (default "/Users/rimas/.helm")
      --host string                     address of Tiller. Overrides $HELM_HOST
      --kube-context string             name of the kubeconfig context to use
      --tiller-connection-timeout int   the duration (in seconds) Helm will wait to establish a connection to tiller (default 300)
      --tiller-namespace string         namespace of Tiller (default "kube-system")

Use "helm [command] --help" for more information about a command. one

Commit:
use dot notation for release candidates

Commit:
setup connection after displaying client version

mhuttner committed:
Fixed error in docs for file access

$path.base gives the error "can't evaluate field base in type string", it should be "base $path"

Commit:
revert back to /tiller

when running `helm init --upgrade`, the entrypoint since v2.9.1 changed which
caused tiller to never start. Kubernetes may be holding onto the same entrypoint
during upgrades.

adamreese committed:
ref(*): kubernetes v1.11 support

adamreese committed:
docs(generated): regenerate markdown docs

rimusz committed:
add support for `auth-provider` from kubeconfig files, addreses #4422

tomasr committed:
Propagate query string arguments in repository URL to absolute chart URL

tomasr committed:
Only propagate query string if refURL is relative to baseURL

novas0x2a committed:
[manifests] hoist the regex out of SplitManifests

Commit:
distribute tiller binary with each release

mattfarina committed:
Updating to the k8s label convention

Closes #4335

Signed-off-by: Matt Farina <matt@mattfarina.com>

Commit:
bump version to v2.10

hickeyma committed:
Add link to doc for Helm Stop plugin

Link added to
https://github.com/helm/helm/blob/master/docs/related.md#helm-plugins

technosophos committed:
docs(alpine): quote release label value (#4460)

Labels should be quoted so that values such as "true" or "1" are not
interpolated to the wrong type.

Commit:
sort links alphabetically

muffin87 committed:
Add basic tutorial for beginners (#4466)

rimusz committed:
add Tillerless Helm blog post and plugin references

aswinkarthik committed:
fix(helm): Add --tiller-tls-hostname flag to 'helm init'

This will fix the bug where 'helm init --wait' fails when TLS is used.

Signed-off-by: aswinkarthik <aswinkarthik93@gmail.com>

ultimateboy committed:
fix(client): fix bug in list releases to append all releases

rosko committed:
fix: link to custom resource definitions section

mattfarina committed:
docs(README): Updating for Helm in CNCF

Signed-off-by: Matt Farina <matt@mattfarina.com>

nickschuch committed:
Update zoom.us link

nojnhuh committed:
Fix typo in parser.go

Commit:
strip out all extra lines other than the first for parsing

Commit:
Improve documentation for helm upgrade (#4030)

Expand and clarify documentation for `helm upgrade` to include nuances
of command line values setting.

Fixes issue #4030

mattfarina committed:
Moving from CLA to DCO in contribution guide

Signed-off-by: Matt Farina <matt@mattfarina.com>

dustinspecker committed:
docs: remove extraneous "it"

hickeyma committed:
Fix helm create note for k8 label convention changes

Update to note as per changes in Issue #4335 . Otherwise cannot
retrieve the pod with the label names.

hobti01 committed:
Messages are encrypted when using TLS

When helm client and tiller use TLS their messages are encrypted, not unencrypted ;)

nojnhuh committed:
Fix typo in message.go

Commit:
Tweeks per bacon review

derkoe committed:
Set proxy for all connections, fixes #4326 (#4579)

Signed-off-by: Christian Koeberl <christian.koeberl@gmail.com>

databus23 committed:
Avoid importing k8s.io/kubernetes from pkg/helm (#4499)

* Avoid importing k8s.io/kubernetes from pkg/helm

When writing a helm client (e.g. a helm plugin) that talks to tiller importing k8s.io/helm/pkg/helm to get the grpc client is key.
This pkg should not have a dependency to the k8s.io/kubernetes to avoid pulling in a lot of code that is only used within tiller and blow up binary sizes.

Signed-off-by: Fabian Ruff <fabian@progra.de>

* Add references to pull request in errors message

Signed-off-by: Fabian Ruff <fabian@progra.de>

* copy helper function from pkg/storage/driver

Signed-off-by: Fabian Ruff <fabian@progra.de>

* Move storage errors to seperate package

Signed-off-by: Fabian Ruff <fabian@progra.de>

* Keep old  error variables for backward compatibility

Signed-off-by: Fabian Ruff <fabian@progra.de>

Commit:
fix(release_server): handle the case when requested values is empty (#4604)

Signed-off-by: Matthew Fisher <matt.fisher@microsoft.com>

Commit:
allow settings TLS flags from environment variables (#4590)

Signed-off-by: Matthew Fisher <matt.fisher@microsoft.com>

Commit:
introduce `helm init --automount-service-account-token` (#4589)

Signed-off-by: Matthew Fisher <matt.fisher@microsoft.com>

ianchen06 committed:
Fix grammer for tests (#4599)

Signed-off-by: Ian Chen <ianchen06@gmail.com>

antonosmond committed:
Be explicit about where occurences of <CHARTNAME> will be replaced in starter charts (#4548)

Signed-off-by: Anton Osmond <antonosmond@gmail.com>

Commit:
4611 - Remove deadlink to deis.com (#4613)

Signed-off-by: tgarlot <thomas@garlot.net>

sarcasticadmin committed:
Fix for checking helm version slice bounds out of range (#4609)

* fix(helm): Use env to locate bash

Leverage '/usr/bin/env bash` for find bash instead of hardcoding
'/bin/bash' since some *nix OSes have it installed elsewhere.
Signed-off-by: Robert James Hernandez <rob@sarcasticadmin.com>

* test(helm): Adding case for versions short flag

When git sha is empty if being built from tarball running 'helm version
--short' should just ignore '--short' since sha is empty. Adding test
to ensure this is the case.

Signed-off-by: Robert James Hernandez <rob@sarcasticadmin.com>

* fix(helm): ignore short flag when sha is empty

Signed-off-by: Robert James Hernandez <rob@sarcasticadmin.com>

Commit:
Fix race condition in `helm list` (#4620)

* Fix race in helm list when partitioning

Problem:
The chunks slice that is passed through the channel is reused for each
partition. This means that encoding the release into a message is racing with
populating the next partition, causing the results to sometimes not fit in the
message, and the release list to be incorrect

Solution:
Allocate a new slice for each partition

Issue #3322

Signed-off-by: Brian Marshall <bmarshall13@users.noreply.github.com>
(cherry picked from commit a0858e29d877b8aa19478133865e45958ff21a9e)

* fix import sorting

Signed-off-by: Matthew Fisher <matt.fisher@microsoft.com>

* ref(release_server_test): use NewReleaseServer()

Signed-off-by: Matthew Fisher <matt.fisher@microsoft.com>

* add unit test for race condition in `helm list`

Signed-off-by: Matthew Fisher <matt.fisher@microsoft.com>

adshmh committed:
feat(helm): output option for helm get values command, allow json and yaml formats (#4596)

Signed-off-by: Arash Deshmeh <adeshmeh@ca.ibm.com>

Commit:
Clearer wording when using pipelines

Signed-off-by: Nenad Stojanovikj <nekk1@live.com>

vermapratyush committed:
Fix type in Values File (#4629)

Fix typo

Signed-off-by: Pratyush Verma <vermapratyush@gmail.com>

helgi committed:
test and build with golang 1.11 (#4637)

linters were failing with: pkg/urlutil/urlutil_test.go:1::warning: file is not gofmted with -s (gofmt)

Signed-off-by: Helgi Þorbjörnsson <helgith@gmail.com>

cpanato committed:
change kubernetes to helm in docs and yaml files when refer the repository (#4640)

Signed-off-by: cpanato <ctadeu@gmail.com>

technosophos committed:
chore: update Sprig to 2.16 (#4652)

Signed-off-by: Matt Butcher <matt.butcher@microsoft.com>

Commit:
fix merge conflicts/linter errors (#4653)

Signed-off-by: Matthew Fisher <matt.fisher@microsoft.com>

stevewolter committed:
Check for err before working with newRelease. (#4630)

Should fix #4585.

Signed-off-by: Steve Wolter <swolter@google.com>

louismunro committed:
Make ping() request a specific image. Add a getTillerPodImage method. (#4622)

Signed-off-by: Louis Munro <lm@louismunro.com>

yuya-takeyama committed:
Add --no-sudo option (#4649)

Signed-off-by: Yuya Takeyama <sign.of.the.wolf.pentagram@gmail.com>

yuya-takeyama committed:
Install tiller as well if it exists (#4648)

* Install tiller as well if it exists

Signed-off-by: Yuya Takeyama <sign.of.the.wolf.pentagram@gmail.com>

* Fix message

ref: https://github.com/helm/helm/pull/4648#discussion_r218229227

Signed-off-by: Yuya Takeyama <sign.of.the.wolf.pentagram@gmail.com>

qiangli committed:
fix(helm): fix selector typo in service template for 'helm create' (#4663)

changed instancelease to instance in service template

closes #4661

Signed-off-by: Qiang Li <liqiang@gmail.com>

calebdelnay committed:
Fix credentials not set for ResolveChartVersion default HTTP client (#4662)

Fixes Issue #4299 and Issue #4445

Signed-off-by: Caleb Delnay <calebdelnay@gmail.com>

Commit:
fix(helm): fix regression with TLS flags/environment variables not being parsed (#4657)

* fix(helm): fix regression with TLS flags/envvars

This change fixes some of the assumptions made in an earlier commit. Helm's TLS flags and environment variables were not respected because they were parsed well before execution (during settings.AddFlagsTLS()), causing erroneous behaviour at runtime. By re-introducing environment.Init(), Helm can properly parse environment variables at the correct time.

One change that had to occur in this PR is the fact that we need to call settings.Init() each time we call settings.AddFlagsTLS(). This is because each command owns its own FlagSet, so we need to parse each flagset to read and propagate the environment variables correctly.

I also noticed that we were maintaining two separate variables for each TLS value. Refactoring out some of the older code to all use the settings object makes the code much cleaner to read and fixes an issue where setting a flag or environment variable would propagate to the settings object, but we'd be reading from tlsEnable.

I've also added some unit tests to ensure this regression doesn't occur again.

Signed-off-by: Matthew Fisher <matt.fisher@microsoft.com>

* fix bug where os.ExpandEnv() on the default value causes differing behaviour

Signed-off-by: Matthew Fisher <matt.fisher@microsoft.com>

* add more context to the TODO/FIXME messages

Signed-off-by: Matthew Fisher <matt.fisher@microsoft.com>

adshmh committed:
fix(helm): fix paths in the ingress template and values file written by helm create

Signed-off-by: Arash Deshmeh <adeshmeh@ca.ibm.com>

nonspecialist committed:
Don't assume that the returned error from the storage driver isn't nil (#3625) (#4627)

Signed-off-by: Colin Panisset <colin.panisset@cevo.com.au>

technosophos committed:
docs(provenance): update explanation of new GnuPG format (#4710)

Signed-off-by: Matt Butcher <matt.butcher@microsoft.com>

adshmh committed:
feat(helm): use apps/v1 as deployment apiVersion (#4725)

Use 'apps/v1' as the apiVersion field for the deployment.yaml written
by the 'helm create' command, rather than the deprecated 'apps/v1beta2'.

Signed-off-by: Arash Deshmeh <adeshmeh@ca.ibm.com>

adamreese committed:
ref(*): kubernetes v1.12 support

Signed-off-by: Adam Reese <adam@reese.io>

florianrusch committed:
Add the .vscode folder to the .helmignore file

Signed-off-by: Florian Rusch <git@florianrusch.de>

RPSeq committed:
Update circleCI badge to more consistent shield.

The shield badge matches the go report and godoc badges.

Signed-off-by: RPSeq <ryan.smith.p@gmail.com>

hickeyma committed:
Fix Slack channel references (#4752)

Signed-off-by: Martin Hickey <martin.hickey@ie.ibm.com>

teresy committed:
The nil check before the range loop is redundant

Signed-off-by: Rijnard van Tonder <hi.teresy@gmail.com>

ultimateboy committed:
fix(tiller): correctly sort PodDisruptionBudget objects before pods that might use them (#4769)

Signed-off-by: Matt Tucker <ultimateboy@gmail.com>

thomasyuan committed:
update rbac.authorization.k8s.io from v1beta1 to v1 (#4771)

Signed-off-by: Thomas Yuan <thomas@cppcoder.net>

etiennetremel committed:
Add Helm convert plugin (#4766)

Signed-off-by: Etienne Tremel <etienne.tremel@container-solutions.com>

Anumita committed:
feat(helm): add $HELM_KEY_PASSPHRASE environment variable for signing helm charts (#4778)

* feat(helm): add $HELM_KEY_PASSPHRASE environment variable for signing helm charts

If $HELM_KEY_PASSPHRASE is set then helm package sign command will not prompt the
user to enter the passphrase for the private key

Signed-off-by: Anumita Shenoy <ansheno@microsoft.com>

* docs(helm):  added documentation for HELM_KEY_PASSPHRASE

Added description for HELM_KEY_PASSPHRASE

Signed-off-by: Anumita Shenoy <ansheno@microsoft.com>

mortent committed:
fix(helm): Use line breaks consistently in status output

The output from helm status does not have consistent use of line breaks.
For some resources there is a line break after the kind header, for
others there is not. This is caused by how the printer handles column
headers. This removes a line break for all but the first resource listed.

Signed-off-by: Morten Torkildsen <mortent@google.com>

hickeyma committed:
Add test template example for helm create (#4524)

* Add test template example for helm create

As helm create ganerates an nginx chart, the example is based on
https://github.com/helm/helm/blob/master/docs/examples/nginx/templates/
service-test.yaml

Signed-off-by: Martin Hickey <martin.hickey@ie.ibm.com>

* Update after review

- https://github.com/helm/helm/pull/4524#pullrequestreview-155183390
- https://github.com/helm/helm/pull/4524#pullrequestreview-155379922
- Fix where app.kubernetes.io/instance was incorrect in service selector

Signed-off-by: Martin Hickey <martin.hickey@ie.ibm.com>

* Add doc update

Signed-off-by: Martin Hickey <martin.hickey@ie.ibm.com>

* Update GO formatting

Signed-off-by: Martin Hickey <martin.hickey@ie.ibm.com>

mgresser committed:
Grammar fix (#4801)

Signed-off-by: Mischa Gresser <mgresser@dstillery.com>

jjasghar committed:
Update quickstart.md

Updated the MySQL release, and put more output that gets sent to
STDOUT with `helm install stable/mysql`

Signed-off-by: JJ Asghar <jja@ibm.com>

marckhouzam committed:
Fix reference to wordpress (#4803)

Signed-off-by: Marc Khouzam <marc.khouzam@ville.montreal.qc.ca>

mortent committed:
fix(helm): Update status output to include resource details (#4791)

Update of the client-go package changed the status output to only include
the age of resources. The new printer in client-go only formats the output
to include details of specific resources if the internal representation
of resources are passed into the printer. This PR updates helm to convert
resources to the internal type before printing.

Closes #4712

Signed-off-by: Morten Torkildsen <mortent@google.com>

AlphaWong committed:
UPT: tutorial (#4808)

Signed-off-by: Alpha <jhexe2002@gmail.com>

marckhouzam committed:
Update Contributing.md to remove mention of CLA (#4804)

* Update Contributing.md to remove mention of CLA

Signed-off-by: Marc Khouzam <marc.khouzam@ville.montreal.qc.ca>

* Add hyperlink to git page in CONTRIBUTING.md

Signed-off-by: Marc Khouzam <marc.khouzam@ville.montreal.qc.ca>

ledor473 committed:
Missing the batch permission in one of the example (#4829)

Signed-off-by: Louis-Etienne Dorval <ledor473@hotmail.com>

masahiro331 committed:
Fix cmd/helm use tillerTunnel values (#4777)

Signed-off-by: masahiro331 <m_fujimura@r.recruit.co.jp>

bartelsielski committed:
Remove newline at the start of zsh completion file (#4851)

Signed-off-by: Bartel Sielski <bsielski@nalys-group.com>

maorfr committed:
Add plugins and tools to docs (#4861)

Signed-off-by: Maor <maor.friedman@nuvo-group.com>

Commit:
bump version to v2.11 (#4700)

Signed-off-by: Matthew Fisher <matt.fisher@microsoft.com>
(cherry picked from commit 3a551d01d8e021679fd3d05ef9b97136e99b79e7)

Commit:
test(tiller): cover crash fixed by #4630 (#4853)

While investigating a tiller crash on v2.10.0 (see recent comments in #3125), I pulled down the code
and wrote a test replicating the crash I was experiencing. I then
realized that the crash had been fixed, and was able to locate the fix
in #4630 after running a quck bisect.

Since there don't appear to be any tests that cover this crash, and I
had written one myself, I figured I might as well put up a PR for it.

Here's what the test failure on v2.10.0 looks like:
```
-- FAIL: TestUpdateReleasePendingInstall_Force (0.00s)
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x50 pc=0x1d128d8]

goroutine 235 [running]:
testing.tRunner.func1(0xc420493c20)
        /usr/local/Cellar/go/1.10/libexec/src/testing/testing.go:742 +0x29d
panic(0x1eb8d80, 0x2a12db0)
        /usr/local/Cellar/go/1.10/libexec/src/runtime/panic.go:505 +0x229
k8s.io/helm/pkg/tiller.(*ReleaseServer).performUpdateForce(0xc4208210b0, 0xc4202c6dc0, 0x0, 0x0, 0x2174220)
        /Users/mattrasmus/go/src/k8s.io/helm/pkg/tiller/release_update.go:166 +0x188
k8s.io/helm/pkg/tiller.(*ReleaseServer).UpdateRelease(0xc4208210b0, 0x2191780, 0xc420820f30, 0xc4202c6dc0, 0x29aeb90, 0x38, 0x2d2)
        /Users/mattrasmus/go/src/k8s.io/helm/pkg/tiller/release_update.go:43 +0x245
k8s.io/helm/pkg/tiller.TestUpdateReleasePendingInstall_Force(0xc420493c20)
        /Users/mattrasmus/go/src/k8s.io/helm/pkg/tiller/release_update_test.go:549 +0x120
testing.tRunner(0xc420493c20, 0x20e5c70)
        /usr/local/Cellar/go/1.10/libexec/src/testing/testing.go:777 +0xd0
created by testing.(*T).Run
        /usr/local/Cellar/go/1.10/libexec/src/testing/testing.go:824 +0x2e0
FAIL    k8s.io/helm/pkg/tiller  0.118s
```

Signed-off-by: Matt Rasmus <mrasmus@betterworks.com>

zachomedia committed:
Remove executable bits from chartutil generated files (#4839)

Signed-off-by: Zachary Seguin <zachary@zacharyseguin.ca>

adamreese committed:
ref(kube): use external api types where possible

Signed-off-by: Adam Reese <adam@reese.io>

Commit:
add s390x architecture to the release notes (#4741)

Signed-off-by: Matthew Fisher <matt.fisher@microsoft.com>

Commit:
docs(release_checklist): fix changelog generation command (#4694)

Signed-off-by: Matthew Fisher <matt.fisher@microsoft.com>

Commit:
add special note for RBAC users (#4597)

Signed-off-by: Matthew Fisher <matt.fisher@microsoft.com>

danielmbarlow committed:
Small typo fix (#4887)

Signed-off-by: Daniel M Barlow <danielmbarlow@gmail.com>

seaneagan committed:
fix(engine): Fix template rendering thread safety issue (#4828)

Remove the engine `currentTemplates` field which was shared state
across threads and thus not thread safe, and instead just pass these
reference templates as parameters down recursively.

Closes #4819

Signed-off-by: Sean Eagan <sean.eagan@att.com>

drnic committed:
Full path to mysubchart file (#4882)

Signed-off-by: Dr Nic Williams <drnicwilliams@gmail.com>

mgaruccio committed:
fix snap install command (#4877)

Added `--classic` to the snap install command.

Signed-off-by: Mike Garuccio <mgaruccio@gmail.com>

timthelion committed:
Document the _proto directory (#4756)

Signed-off-by: Timothy Hobbs <timothy@hobbs.cz>

rimusz committed:
make get script sudo optional (#4677)

Signed-off-by: rimas <rmocius@gmail.com>

adshmh committed:
fix(helm): fix incorrect yaml output format of get hooks command (#4684)

Signed-off-by: Arash Deshmeh <adeshmeh@ca.ibm.com>

jianliao82 committed:
fix:#4873, check release name (#4883)

for command helm template and helm install
when cusotmer input invalid release name, return a error

Signed-off-by: jliao <jliao@alauda.io>

consideRatio committed:
Reform indentation practices using nindent (#4562)

The essence of this commit is to help people get started with a better
indentation practice than this:

```yaml
spec:
  labels:
{{ toYaml .Values.labels | indent 4 }}
```

The previous indentation practice is harder to read. Instead this commit
introduces an indentation practice using `nindent` like this:

```yaml
spec:
  labels:
    {{- toYaml .Values.labels | nindent 4 }}
```

Signed-off-by: Erik Sundell <erik.i.sundell@gmail.com>

adshmh committed:
include the name of the missing object in release uninstall error (#4635)

* fix(tiller) added mock kube client to return failure on delete

Signed-off-by: Arash Deshmeh <adeshmeh@ca.ibm.com>

* fix(tiller) added object's name to delete release error message

Signed-off-by: Arash Deshmeh <adeshmeh@ca.ibm.com>

maorfr committed:
related docs - remove import, export, add backup and plugin-utils instead (#4867)

* remove import, export, add backup instead

Signed-off-by: Maor <maor.friedman@nuvo-group.com>

* add helm plugin utility functions

Signed-off-by: Maor <maor.friedman@nuvo-group.com>

* move helm-plugin-utils into helm plugins section

Signed-off-by: Maor <maor.friedman@nuvo-group.com>

ryandawsonuk committed:
note on prefixing of resource names with release name (#4633)

* note on naming of resources

Signed-off-by: Ryan Dawson <ryan.dawson@alfresco.com>

* put whitespace back in

Signed-off-by: Ryan Dawson <ryan.dawson@alfresco.com>

* put whitespace back

Signed-off-by: Ryan Dawson <ryan.dawson@alfresco.com>

* put whitespace back

Signed-off-by: Ryan Dawson <ryan.dawson@alfresco.com>

* clarify template fullname

Signed-off-by: Ryan Dawson <ryan.dawson@alfresco.com>

* fix formatting problem by escaping underscore

Signed-off-by: ryandawsonuk <ryandawson@cantab.net>

* no need to change whitespace elsewhere in doc

Signed-off-by: ryandawsonuk <ryandawson@cantab.net>

Commit:
s,kubernetes/charts,helm/charts,g (#4889)

Signed-off-by: Matthew Fisher <matt.fisher@microsoft.com>

Commit:
add appveyor for testing on Windows (#4136)

Signed-off-by: Matthew Fisher <matt.fisher@microsoft.com>

mortent committed:
fix(helm): Merge nested values correctly on upgrade (#4806)

Upgrading a release and override existing values doesn't work as expected for nested values. Maps should be merged recursively, but currently maps are treated just like values and replaced at the top level.

If the existing values are:
```yaml
resources:
  requests:
    cpu: 400m
  something: else
```
and an update is done with ```--set=resources.requests.cpu=500m```, it currently ends up as
```yaml
resources:
  requests:
    cpu: 500m
```
but it should have been
```yaml
resources:
  requests:
    cpu: 500m
  something: else
```

This PR updates the way override values are merged into the existing set of values to merge rather than replace maps.

Closes: #4792

Signed-off-by: Morten Torkildsen <mortent@google.com>

bmperrea committed:
fix(tiller): rollback deleted release (#3722) (#4820)

Solves #3722 by making the changes in #3539 more compatible with the previous behavior.

This gives a recovery option for "oops I deleted my helm release" by allowing rollback, which is intended to be a working feature of helm. Note that purging releases removes the history required to rollback, so this doesn't work in that case. However, purging takes significantly more time, so it's harder to accidentally purge everything.

Signed-off-by: Brent <bmperrea@gmail.com>

leominov committed:
fix(helm): Non-zero exit code on failed chart repository update (#4348)

* Fail on failed chart repository update

* Fix failed test

* Add repo update strict flag

scottrigby committed:
Helm code of conduct (#4901)

See https://github.com/helm/chart-testing/issues/48

Signed-by: Scott Rigby <scott@r6by.com>
Signed-off-by: Scott Rigby <scott@r6by.com>

Commit:
fix(docs): run `make docs` (#4924)

Signed-off-by: Matthew Fisher <matt.fisher@microsoft.com>

mortent committed:
fix(helm): Fix linebreaks when printing custom resources

The output from helm status is not correct for custom resources. The
HumanReadablePrinter from Kubernetes only outputs the column names when
the type differs from the previous one. This makes the output
inconsistent and also creates problems for putting in the correct line
breaks. This PR sets up a new printer for each type, thereby making sure
that all types are printed with the correct use of line breaks and with
column names.

Signed-off-by: Morten Torkildsen <mortent@google.com>

mortent committed:
fix(helm): Print details for pod resource

Due to a regression from a previous change, details about pod resources
does not show up in the status output. This makes sure that the pod type
from core are passed in to the printer so the details are shown in the
output.

Signed-off-by: Morten Torkildsen <mortent@google.com>

danjwinter committed:
fix(helm): --set for out of order list values (#4682)

When a user specifies value overrides for list values out of order,
strvals.listItem panics. Change strvals.listItem to handle this case by
re-initializing nil values to a new map.

Closes #4503

Co-authored-by: Cameron Childress <cameron@cchildress.org>
Co-authored-by: Kevin Collette <hal.collette@gmail.com>
Co-authored-by: Connor McKelvey <connormckelvey@gmail.com>
Co-authored-by: Dan Winter <dan.j.winter@gmail.com>

Signed-off-by: Dan Winter <dan.j.winter@gmail.com>
Signed-off-by: Cameron Childress <cameron@cchildress.org>
Signed-off-by: Kevin Collette <hal.collette@gmail.com>
Signed-off-by: Connor McKelvey <connormckelvey@gmail.com>

jolson490 committed:
Securing tiller via running it locally. (#4869)

* Securing tiller via running it locally.

Signed-off-by: Joshua Olson <joshua.olson.490@gmail.com>

* Clarify that TLS is the recommended way to secure Helm.

Signed-off-by: Joshua Olson <joshua.olson.490@gmail.com>

glb committed:
docs: use nindent in documentation examples (#4905)

* docs: markdown prettyprinting

Signed-off-by: Geoff Baskwill <me@geoffbaskwill.ca>

* docs: use nindent in documentation examples

Fixes #4890.

Signed-off-by: Geoff Baskwill <me@geoffbaskwill.ca>

filipre committed:
Update control_structures.md (#4948)

update position of `and` operator

Signed-off-by: René Filip <rene.filip@sap.com>

glb committed:
chore(deps): update cobra to latest (#4913)

Signed-off-by: Geoff Baskwill <me@geoffbaskwill.ca>

Commit:
fix(windows): fix unit tests on Windows (#4897)

Signed-off-by: Matthew Fisher <matt.fisher@microsoft.com>

tariq1890 committed:
Adding link labels to doc links for more conspicuity (#4904)

Signed-off-by: tariqibrahim <tariq.ibrahim@microsoft.com>

tao12345666333 committed:
Fix doc charts indent (#4940)

Signed-off-by: Jintao Zhang <zhangjintao9020@gmail.com>

a8uhnf committed:
Error message if anyone try to create multiple helm chart at a time (#4952)

* error message for creating multiple helm chart at a time

Signed-off-by: hanif <abuhanifa@pathao.com>

* Update cmd/helm/create.go

Co-Authored-By: a8uhnf <a8uhnf@users.noreply.github.com>
Signed-off-by: hanif <abuhanifa@pathao.com>

OrthoDex committed:
docs: add documentation for the helmignore file (#4966)

* docs: add documentation for the helmignore file

Signed-off-by: Ishaan Malhi <ishaan.malhi@gmail.com>

* docs: rearrange helmignore docs under chart template developer's guide

Signed-off-by: Ishaan Malhi <ishaan.malhi@gmail.com>

hickeyma committed:
Update chart doc with details about the license file (#4977)

Signed-off-by: Martin Hickey <martin.hickey@ie.ibm.com>

ultimateboy committed:
fix(storage): when pruning release versions, never delete the last deployed revision

Signed-off-by: Matt Tucker <ultimateboy@gmail.com>

lukehoban committed:
Allow missing trailing '/' in --repo url (#4956)

Apply the same procedure to allow missing trailing slash in repo base URLs used in `repo/chart` inputs to `--repo` inputs.

Fixes #4954.

Signed-off-by: Luke Hoban <luke@pulumi.com>

tariq1890 committed:
avoid kubernetes import for slice contains logic (#4963)

* avoid kubernetes import for slice contains logic

Signed-off-by: tariqibrahim <tariq.ibrahim@microsoft.com>

* fix review comments

Signed-off-by: tariqibrahim <tariq.ibrahim@microsoft.com>

Commit:
Fix for existing CRDs are deleted when crd-install hook is introduced (#4709)

Signed-off-by: mmerrill3 <michael.merrill@vonage.com>

mortent committed:
fix(helm): Allow custom resources in hooks (#4986)

Currently the code that handles hooks uses a builder that creates the versioned types rather than unstructured. This results in an error whenever a custom resource is used in the hook as the type will not be registered in the scheme used in Helm. This changes this to use a builder that created unstructured resources and only converts to the versioned type when needed.

Signed-off-by: Morten Torkildsen <mortent@google.com>

hickeyma committed:
Update messaging in value parsing to improve traceability (#4974)

Closes #4736

Signed-off-by: Martin Hickey <martin.hickey@ie.ibm.com>

tariq1890 committed:
add unit tests for portforwarder (#4979)

Signed-off-by: tariqibrahim <tariq.ibrahim@microsoft.com>

EppO committed:
Charts docs clean-up

- Address #4998, #5000, #5001 and #5002
- Fix Markdown syntax for code blocks

Signed-off-by: Florent Monbillard <f.monbillard@gmail.com>

retr0h committed:
Added extra padding to resources template (#4981)

This is probably a silly PR.  However, each time I create a new chart
via `helm create`, the instructions to uncomment the resource limits
require adding a couple extra spaces back to the YAML.

This PR simply brings the spacing in-line with the rest of the generated
template.

Signed-off-by: John Dewey <john@dewey.ws>

thomastaylor312 committed:
ref(kube): Gets rid of superfluous Sprintf call

Signed-off-by: Taylor Thomas <thomastaylor312@gmail.com>

henrynash committed:
docs(helm): Mention commit signing in the Developer Guide (#5018)

This change adds a mention to the reqruiement for commit signing
to the Git Convention outline in the Developer Guide, while
leaving the existing detail on signing in the Contributing doc.

Closes #5016

Signed-off-by: Henry Nash <henry.nash@uk.ibm.com>

henrynash committed:
fix(helm): get rid of lint warning in pkg/storage (#5021)

Provide comment for constant to avoid golint warning.

Closes #5020

Signed-off-by: Henry Nash <henry.nash@uk.ibm.com>

henrynash committed:
fix(helm): Correct and improve resilence of template check in unit test (#5010)

Make the current check for the number of templates on create more
resilient by using a varible to store the expected number
of templates. In addition, fix the error message so that it displays
the correct number expected of templates.

Closes #5009

Signed-off-by: Henry Nash <henry.nash@uk.ibm.com>

helgi committed:
Return empty string instead of nil when linting on required (#4748)

* Return empty string instead of nil when linting on required

This allows lint to work in scenarios when required is used in secrets or it's output is passed to another function.
Due to lint mode no longer failing on missing value in required it is passing nil through which not all functions can accept.

Fixes #4747

Signed-off-by: Helgi Þorbjörnsson <helgith@gmail.com>

* Apply suggestions from code review

Co-Authored-By: helgi <70530+helgi@users.noreply.github.com>
Signed-off-by: Helgi Þorbjörnsson <helgith@gmail.com>

* Add tests

Signed-off-by: Helgi Þorbjörnsson <helgith@gmail.com>

bacongobbler commented:
If @thomastaylor312 says it's fine then I trust him. The rest of the code looks great :) @splisson if you wouldn't mind rebasing this PR, we can get it merged for 2.13.0 (the next release after KubeCon). Let me know if you need help!

frankh committed:
Add --parallel flag to helm test (#4144)

* Refactor test run to separate method

This will allow us to parallelise it more easily

Signed-off-by: Frank Hamand <frankhamand@gmail.com>

* Add --parallel flag to helm test

(No functionality in this commit)

Signed-off-by: Frank Hamand <frankhamand@gmail.com>

* Run helm tests in parallel with --parallel flag

Signed-off-by: Frank Hamand <frankhamand@gmail.com>

* Add a mutex to helm test message streams

This is to protect against data races when running tests in parallel.

Signed-off-by: Frank Hamand <frankhamand@gmail.com>

* Add tests for --parallel flag

Signed-off-by: Frank Hamand <frankhamand@gmail.com>

* Add concurrency limit for parallel helm tests

Signed-off-by: Frank Hamand <frankhamand@gmail.com>

* Add test for concurrency limit

Signed-off-by: Frank Hamand <frankhamand@gmail.com>

* Fix rebase introduced errors

Signed-off-by: Frank Hamand <frankhamand@gmail.com>

splisson-altair commented:
I'm on it

splisson committed:
fix comment

splisson committed:
Merge branch 'bug/#3020' of github-splisson:splisson/helm into bug/#3020

Signed-off-by: Sebastien Plisson <sebastien.plisson@gmail.com>

splisson commented:
Damn, I tried to resolve the DCO issue but following the DCO instruction ended up signing off all commits... trying to revert that

splisson commented:
I'll try to put my changes in a new branch and make another PR, that should make things cleaner

splisson commented:
I opened a new PR with just my changes on top of the latest code from master https://github.com/helm/helm/pull/5024

splisson commented:
@bacongobbler We can probably close this PR and include the new one #5024 in the release instead
